### ISSUE TITLE
echidna fails to run outside the repository folder

### ISSUE THREAD

**Issue by makoto** [trailofbits/echidna#10]
Tried the example solution @japesinator suggested during his ETHCC workshop but does not seem working on the latest solc compiler.

```
$  echidna-test contracts/testme.sol 
━━━ contracts/testme.sol ━━━
  ○ 0/1 complete (running)
  ↯ "echidna_alwaystrue" failed after 1 test (shrinking)
echidna-test: lib/Echidna/Exec.hs: openFile: does not exist (No such file or directory)
```

```
pragma solidity 0.4.21;

contract Canal {
  bool private first_gate_up=false;
  bool private second_gate_up=true;

  function lower(bool first_gate) public returns (bool){
    if(first_gate) {
      if(second_gate_up) {
        first_gate_up = false;
        return(true);
      } else {
        return(false);
      }
    } else {
      if(first_gate_up) {
        second_gate_up = false;
        return true;
      } else {
        return(false);
      }
    }
  }
  function raise(bool first_gate) public {
    if(first_gate) {
      first_gate_up = true;
    } else {
      second_gate_up = true;
    }
  }

  function echidna_alwaystrue() public view returns (bool){
    return (first_gate_up || first_gate_up);
  }
}
```

**Comment by ggrieco-tob** [comment]
I cannot reproduce exactly your error:

```bash
$ solc --version
solc, the solidity compiler commandline interface
Version: 0.4.21+commit.dfe3193c.Linux.g++
$ echidna-test contracts/testme.sol 
━━━ contracts/testme.sol ━━━
  ✗ "echidna_alwaystrue" failed after 1 test and 1 shrink.
  
      │ Call sequence: lower(true);
  
  ✗ 1 failed.
```

**Comment by snd** [comment]
i get the same error with solc 0.4.18

```
$ solc --version
solc, the solidity compiler commandline interface
Version: 0.4.18+commit.9cf6e910.Darwin.appleclang
$ echidna-test contracts/testme.sol 
━━━ contracts/testme.sol ━━━
  ○ 0/1 complete (running)
  ↯ "echidna_alwaystrue" failed after 1 test and 1 shrink (shrinking)
echidna-test: lib/Echidna/Exec.hs: openFile: does not exist (No such file or directory)
```

OSX 10.11.6

built from https://github.com/trailofbits/echidna/commit/159b595172e85e67b90ba374c253a40e32c1f568

installed as described there: https://github.com/trailofbits/echidna/issues/12#issuecomment-372033419

**Comment by snd** [comment]
the above was in a directory different from the echidna directory.

when i copy the contract into the echidna directory (where i cloned and installed with stack) it works:

```
$ echidna-test testme.sol 
━━━ testme.sol ━━━
  ✗ "echidna_alwaystrue" failed after 3 tests.
  
      │ Call sequence: lower(true);
  
  ✗ 1 failed.
```

it should work in any directory

**Comment by makoto** [comment]
Yeah, I created the test sol file on a brand new project I created using truffle.

**Comment by ggrieco-tob** [comment]
I can confirm that echidna fails to run properly outside the repository folder. Sorry for the long delay. It seems to be [a known bug in hedgehog](https://github.com/hedgehogqa/haskell-hedgehog/issues/105).

**Comment by ggrieco-tob** [comment]
Please test this possible fix: https://github.com/trailofbits/echidna/pull/29

**Comment by ggrieco-tob** [comment]
Should be fixed in 75afb17.

**Comment by thumphries** [comment]
This was also fixed upstream in Hedgehog recently, sorry about that! Is part of the 0.6 release. https://github.com/hedgehogqa/haskell-hedgehog/pull/184
